tket/src/Transformations/OptimisationPass.cpp

```diff
@@ -37,11 +40,23 @@ Transform peephole_optimise_2q() {
       synthesise_tket());
 }
 
-Transform full_peephole_optimise(bool allow_swaps) {
-  return (
-      synthesise_tket() >> two_qubit_squash() >> clifford_simp(allow_swaps) >>
-      synthesise_tket() >> three_qubit_squash() >> clifford_simp(allow_swaps) >>
-      synthesise_tket());
+Transform full_peephole_optimise(bool allow_swaps, OpType target_2qb_gate) {
+  switch (target_2qb_gate) {
+    case OpType::CX:
+      return (
+          synthesise_tket() >> two_qubit_squash() >>
+          clifford_simp(allow_swaps) >> synthesise_tket() >>
+          three_qubit_squash() >> clifford_simp(allow_swaps) >>
+          synthesise_tket());
+    case OpType::TK2:
+      return (
+          synthesise_tk() >> two_qubit_squash(OpType::TK2) >>
+          clifford_simp(allow_swaps) >> synthesise_tk() >>
```

Comment:
This should take care of Cliffords properly: any `TK2` gate that was clifford and decomposed will be resynthesised after `clifford_simp`. Swaps should be swallowed within TK2s and be handled later.

```suggestion
          clifford_simp(false) >> two_qubit_squash(OpType::TK2) >> synthesise_tk() >>
```